google-cloud-pubsub/pom.xml

```diff
@@ -147,6 +150,7 @@
             <ignoredUnusedDeclaredDependencies>
               <ignoredUnusedDeclaredDependency>io.opencensus:opencensus-impl</ignoredUnusedDeclaredDependency>
               <ignoredUnusedDeclaredDependency>javax.annotation:javax.annotation-api</ignoredUnusedDeclaredDependency>
+              <ignoredUnusedDeclaredDependency>com.google.auth:google-auth-library-oauth2-http</ignoredUnusedDeclaredDependency>
```

Comment:
Somehow I have to mark this artifact in ignoredUnusedDeclaredDependency even though it's actually used. 

<img width="1168" alt="Screen Shot 2021-10-18 at 4 54 11 PM" src="https://user-images.githubusercontent.com/28604/137805255-d321df19-843e-49cd-85f5-3a9168919597.png">

Reply:
I think we won't need to add an exception when compilation becomes really broken after the new release of the shared dependencies BOM.